lib/vrl/stdlib/src/parse_ruby_hash.rs

```diff
@@ -109,36 +140,57 @@ fn parse_nil<'a, E: ParseError<&'a str>>(input: &'a str) -> IResult<&'a str, Val
 fn parse_bytes<'a, E: HashParseError<&'a str>>(input: &'a str) -> IResult<&'a str, Bytes, E> {
     context(
         "bytes",
-        map(
-            preceded(char('\"'), cut(terminated(parse_str::<'a, E>, char('\"')))),
-            |value| Bytes::copy_from_slice(value.as_bytes()),
-        ),
+        map(alt((parse_str('"'), parse_str('\''))), |value| {
+            Bytes::copy_from_slice(value.as_bytes())
+        }),
     )(input)
 }
 
+fn parse_simple_key<T, E: ParseError<T>>(input: T) -> IResult<T, T, E>
+where
+    T: InputTakeAtPosition,
+    <T as InputTakeAtPosition>::Item: AsChar,
+{
+    input.split_at_position1_complete(
+        |item| {
+            let c = item.as_char();
+            !c.is_alphanum() && c.as_char() != '_'
```

Comment:
Looks like symbol keys in ruby hashes don't allow starting with a digit:

```bash
$ irb
irb(main):001:0> foo = { 1: 123 }
Traceback (most recent call last):
        3: from /usr/bin/irb:23:in `<main>'
        2: from /usr/bin/irb:23:in `load'
        1: from /Library/Ruby/Gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
SyntaxError ((irb):1: syntax error, unexpected ':', expecting =>)
foo = { 1: 123 }
         ^
```

Starting with a character or underscore is fine

Reply:
Weird, `foo = { 1 => 123 }` seems to be valid, though. (Note the `=>` vs `:`)

Reply:
Ah, the comment from @JeanMertz contains the explanation: https://github.com/timberio/vector/pull/8956#discussion_r698425097.

The part before `=>` is evaluated, so `1` resolves to the literal value `1` and is not a symbol.

Reply:
to follow @JeanMertz 's comment `since we're not an actual Ruby parser`, should we care about `{ 1: 234 }`? If it's not valid, then we should not have it. If we have it, we'll have `{ "1": "234" }` 🤷

Reply:
I think my explanation was not clear enough 😅 
Vector is not a Ruby parser but a log parser (in that case). So should we really consider all the cases that it shouldn't parse considering those should not happen? If a ruby program generates `{ foo => "bar" }` should we throw an error or parse it? Would this even happen?
Should we increase the code complexity of the code base and handle all those little cases that should probably not appear in the logs?
IMO, we should parse it, even if Ruby doesn't, `{ foo => "bar" }` generating the `{ "foo": "bar" }` equivalent.
@JeanMertz does this seem ok with you?